mplc/doc/documentation.md

```diff
@@ -0,0 +1,497 @@
+
+   - `["PVRL"]` **Partner Valuation by Reinforcement Learning** 
+    
+- **Federated step-by-step**:
+
+    Federated step by step contributivity methods measure the performance variation on the global validation dataset after each minibatch training - These methods give an estimation on how the model improved on every node.
+    The methods are best suited for federated averaging learning.
+    For each computation round, the contributivity for each partner is calculated as the ratio between the validation score of the newly trained model for each partner and the validation score from the previously trained collective model.
+    Initial rounds (10%) and final rounds (10%) are discarded from calculation as performance increments from the first minibatches might be huge and increments form the last minibatches might be very noisy. Discarded proportions are for now set in the code.
+
+    3 contributivity methods are proposed to adjust the importance of last computation rounds compared to the first ones:
+    - `["Federated SBS linear"]` - Linear importance increase between computation rounds (1000th round weights 1000 times first round)
+    - `["Federated SBS quadratic"]` - Quadratic importance increase between computation rounds (1000th round weights 10e6 times first round)
+    - `["Federated SBS constant"]`- Constant importance increase between computation rounds (1000th round weights same as first round)
+    
+- **Label Flipping**
+    
+    Label Flipping method provides a way to detect mislabelled datasets.
+    The main idea is, while training the model, to learn the probability of a label to be flipped in another, inside each partner dataset. 
+    Then, we flip the label of the noisy data to the most likely right label and train the main model on these likely right data.  
+    A contributivity measure can be inferred from partner's matrices of flip-probability, by computing the exponential inverse of the Frobenius distance to the identity. 
+    However this measure is to be handle carefully, the method is not designed specifically for contributivity measurement, but for mislabelled dataset detection. 
+    
+    - `["LFlip"]` - Label flipping method
+
+> Note: when `methods` is omitted in the config file only the distributed learning is run.
+
+Example: `methods=["Shapley values", "Independent scores", "TMCS"]`
+
+### Miscellaneous
+
+- `is_quick_demo`: `True` or `False` (default)  
+  When set to `True`, the amount of data samples and the number of epochs and mini-batches are significantly reduced, to minimize the duration of the run. This is particularly useful for quick demos or simply for debugging.  
+  Example: `is_quick_demo=True`
+
+## Dataset generation
+
+The `Dataset` object is useful if you want to define custom datasets and related objects such as preprocessing functions and model generator.
+
+### Dataset
+
```

Comment:
The whole section is obsolete

Reply:
Updated

Reply:
@arthurPignet I let you close this conversation or amend/complement the text directly